qlkube/src/decorateBaseSchema.js

```diff
@@ -0,0 +1,65 @@
+const { gql } = require('apollo-server-core');
+const { extendSchema } = require('graphql/utilities');
+const { addResolversToSchema } = require('@graphql-tools/schema');
+
+/**
+ *
+ * @param {*} targetQuery : Query wrapped int the extended object Type
+ * @param {*} extendedType : Object type that should be extended
+ * @param {*} argsNeeded : Optional arguments extracted from the father and passed from the wrapper
+ * @param {*} baseSchema : GraphQL Schema where the type and query are written
+ * @param {*} nameWrapper : Optional custom name for the wrapper
+ * @returns
+ */
+
+function capitalizeType(name) {
+  return name[0].toUpperCase() + name.slice(1);
+}
+
+module.exports = {
+  decorateBaseSchema: function (
+    targetQuery,
+    extendedType,
+    argsNeeded,
+    nameWrapper,
+    baseSchema
+  ) {
+    if (!targetQuery) return baseSchema;
+    if (!extendedType) return baseSchema;
+    const targetType = baseSchema.getQueryType().getFields()[targetQuery];
+    if (!targetType) return baseSchema;
+
+    nameWrapper = nameWrapper ? nameWrapper : 'fieldWrapper';
```

Comment:
I would provide an error in case the `nameWrapper` is not defined, as you did with `targetQuery`